plugins/modules/zos_apf.py

```diff
@@ -508,7 +511,12 @@ def main():
     except ValueError as err:
         module.fail_json(msg="Parameter verification failed", stderr=str(err))
 
-    library = parsed_args.get('library')
+# ### here  vvvv
+# Change to how 'library' is loaded, so we can escape what needs escaping
+#    library = parsed_args.get('library')
+    _orig_library = parsed_args.get("library")
+    library = DataSet.escape_data_set_name(_orig_library)
```

Comment:
You can also use 

```python
data_set = MVSDataSet(name = library)
```

During the class initialization it will automatically escape and resolve the gds relative name if any.